### ISSUE TITLE
Failed to restore nuget since NuGetCommand version 2.146.0 - self signed certificate in certificate chain

### ISSUE THREAD

**Issue by KamilKh88** [Microsoft/azure-pipelines-tasks#9455]
## Environment
- Server - Azure Pipelines
- Agent - Private: 
    -Agent version: Current agent version: **'2.144.2'**
    -OS:  Windows Server 2012 R2 Standard

## Issue Description

Failed to restore nuget since NuGetCommand version 2.146.0 - self signed certificate in certificate chain
![image](https://user-images.githubusercontent.com/47179361/51981907-20c8b280-249d-11e9-9cdf-8bea823a9259.png)

### Task logs (system.debug enabled)

```
2019-01-31T11:58:58.0188979Z ##[debug]Evaluating condition for step: 'NuGet restore'
2019-01-31T11:58:58.0189640Z ##[debug]Evaluating: succeeded()
2019-01-31T11:58:58.0189761Z ##[debug]Evaluating succeeded:
2019-01-31T11:58:58.0189970Z ##[debug]=> True
2019-01-31T11:58:58.0190221Z ##[debug]Result: True
2019-01-31T11:58:58.0190422Z ##[section]Starting: NuGet restore
2019-01-31T11:58:58.0257487Z ==============================================================================
2019-01-31T11:58:58.0257650Z Task         : NuGet
2019-01-31T11:58:58.0257710Z Description  : Restore, pack, or push NuGet packages, or run a NuGet command. Supports NuGet.org and authenticated feeds like Package Management and MyGet. Uses NuGet.exe and works with .NET Framework apps. For .NET Core and .NET Standard apps, use the .NET Core task.
2019-01-31T11:58:58.0257771Z Version      : 2.146.0
2019-01-31T11:58:58.0257840Z Author       : Microsoft Corporation
2019-01-31T11:58:58.0257896Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkID=613747)
2019-01-31T11:58:58.0257994Z ==============================================================================
2019-01-31T11:58:58.0426005Z ##[debug]tf vc resolvePath $\*****.sln /loginType:OAuth /login:.,*** /noprompt
2019-01-31T11:58:58.4732632Z ##[debug]D:\Agents\agent2\_work\7\s\***.sln
2019-01-31T11:58:58.9557489Z ##[debug]agent.TempDirectory=D:\Agents\agent2\_work\_temp
2019-01-31T11:58:58.9603818Z ##[debug]loading inputs and endpoints
2019-01-31T11:58:58.9608887Z ##[debug]loading ENDPOINT_AUTH_PARAMETER_SYSTEMVSSCONNECTION_ACCESSTOKEN
2019-01-31T11:58:58.9618494Z ##[debug]loading ENDPOINT_AUTH_SCHEME_SYSTEMVSSCONNECTION
2019-01-31T11:58:58.9621622Z ##[debug]loading ENDPOINT_AUTH_SYSTEMVSSCONNECTION
2019-01-31T11:58:58.9623521Z ##[debug]loading INPUT_ALLOWPACKAGECONFLICTS
2019-01-31T11:58:58.9625118Z ##[debug]loading INPUT_COMMAND
2019-01-31T11:58:58.9626509Z ##[debug]loading INPUT_CONFIGURATIONTOPACK
2019-01-31T11:58:58.9628046Z ##[debug]loading INPUT_DISABLEPARALLELPROCESSING
2019-01-31T11:58:58.9629754Z ##[debug]loading INPUT_INCLUDENUGETORG
2019-01-31T11:58:58.9631201Z ##[debug]loading INPUT_INCLUDEREFERENCEDPROJECTS
2019-01-31T11:58:58.9632481Z ##[debug]loading INPUT_INCLUDESYMBOLS
2019-01-31T11:58:58.9633822Z ##[debug]loading INPUT_NOCACHE
2019-01-31T11:58:58.9635665Z ##[debug]loading INPUT_NUGETCONFIGPATH
2019-01-31T11:58:58.9637248Z ##[debug]loading INPUT_NUGETFEEDTYPE
2019-01-31T11:58:58.9638951Z ##[debug]loading INPUT_OUTPUTDIR
2019-01-31T11:58:58.9641861Z ##[debug]loading INPUT_PACKTIMEZONE
2019-01-31T11:58:58.9643336Z ##[debug]loading INPUT_PUBLISHPACKAGEMETADATA
2019-01-31T11:58:58.9644868Z ##[debug]loading INPUT_REQUESTEDMAJORVERSION
2019-01-31T11:58:58.9646595Z ##[debug]loading INPUT_REQUESTEDMINORVERSION
2019-01-31T11:58:58.9648023Z ##[debug]loading INPUT_REQUESTEDPATCHVERSION
2019-01-31T11:58:58.9649443Z ##[debug]loading INPUT_SEARCHPATTERNPACK
2019-01-31T11:58:58.9650895Z ##[debug]loading INPUT_SEARCHPATTERNPUSH
2019-01-31T11:58:58.9652134Z ##[debug]loading INPUT_SELECTORCONFIG
2019-01-31T11:58:58.9653339Z ##[debug]loading INPUT_SOLUTION
2019-01-31T11:58:58.9654489Z ##[debug]loading INPUT_TOOLPACKAGE
2019-01-31T11:58:58.9655657Z ##[debug]loading INPUT_VERBOSITYPACK
2019-01-31T11:58:58.9657016Z ##[debug]loading INPUT_VERBOSITYPUSH
2019-01-31T11:58:58.9658343Z ##[debug]loading INPUT_VERBOSITYRESTORE
2019-01-31T11:58:58.9659721Z ##[debug]loading INPUT_VERSIONINGSCHEME
2019-01-31T11:58:58.9663472Z ##[debug]loaded 28
2019-01-31T11:58:58.9676804Z ##[debug]Agent.ProxyUrl=undefined
2019-01-31T11:58:58.9677762Z ##[debug]Agent.CAInfo=undefined
2019-01-31T11:58:58.9678163Z ##[debug]Agent.ClientCert=undefined
2019-01-31T11:58:58.9678530Z ##[debug]Agent.SkipCertValidation=undefined
2019-01-31T11:58:59.0501091Z ##[debug]Agent.ProxyUrl=undefined
2019-01-31T11:58:59.0502753Z ##[debug]check path : D:\Agents\agent2\_work\_tasks\NuGetCommand_333b11bd-d341-40d9-afcf-b32d5ce6f23b\2.146.0\node_modules\vsts-task-tool-lib\lib.json
2019-01-31T11:58:59.0503986Z ##[debug]adding resource file: D:\Agents\agent2\_work\_tasks\NuGetCommand_333b11bd-d341-40d9-afcf-b32d5ce6f23b\2.146.0\node_modules\vsts-task-tool-lib\lib.json
2019-01-31T11:58:59.0504214Z ##[debug]system.culture=en-US
2019-01-31T11:59:00.0458309Z ##[debug]check path : D:\Agents\agent2\_work\_tasks\NuGetCommand_333b11bd-d341-40d9-afcf-b32d5ce6f23b\2.146.0\task.json
2019-01-31T11:59:00.0459054Z ##[debug]adding resource file: D:\Agents\agent2\_work\_tasks\NuGetCommand_333b11bd-d341-40d9-afcf-b32d5ce6f23b\2.146.0\task.json
2019-01-31T11:59:00.0459333Z ##[debug]system.culture=en-US
2019-01-31T11:59:00.0554308Z ##[debug]Getting NuGet
2019-01-31T11:59:00.0555178Z ##[debug]NuGetExeToolPath=D:\Agents\agent2\_work\_tool\NuGet\4.7.0\x64\nuget.exe
2019-01-31T11:59:00.2332138Z ##[debug]nugetfeedtype=internal
2019-01-31T11:59:00.2333008Z ##[debug]externalendpoints=null
2019-01-31T11:59:00.2335596Z ##[debug]command=restore
2019-01-31T11:59:00.2335940Z ##[debug]NuGetExeToolPath=D:\Agents\agent2\_work\_tool\NuGet\4.7.0\x64\nuget.exe
2019-01-31T11:59:00.2336308Z ##[debug]NuGetExeCustomLocation=undefined
2019-01-31T11:59:00.2338286Z ##[debug]searchPatternPack=D:\Agents\agent2\_work\7\s\**\*.csproj
2019-01-31T11:59:00.2339841Z ##[debug]configurationToPack=release
2019-01-31T11:59:00.2341163Z ##[debug]versioningScheme=off
2019-01-31T11:59:00.2343002Z ##[debug]includeReferencedProjects=false
2019-01-31T11:59:00.2344258Z ##[debug]versioningScheme=off
2019-01-31T11:59:00.2345417Z ##[debug]requestedMajorVersion=1
2019-01-31T11:59:00.2346446Z ##[debug]requestedMinorVersion=0
2019-01-31T11:59:00.2347489Z ##[debug]requestedPatchVersion=0
2019-01-31T11:59:00.2348513Z ##[debug]packTimezone=utc
2019-01-31T11:59:00.2348797Z ##[debug]buildProperties=null
2019-01-31T11:59:00.2349099Z ##[debug]basePath=null
2019-01-31T11:59:00.2351650Z ##[debug]verbosityPack=Detailed
2019-01-31T11:59:00.2353433Z ##[debug]includeSymbols=false
2019-01-31T11:59:00.2353973Z ##[debug]NuGet.UseLegacyFindFiles=undefined
2019-01-31T11:59:00.2354219Z ##[debug]NuGetTasks.IsHostedTestEnvironment=undefined
2019-01-31T11:59:00.2354461Z ##[debug]System.TeamFoundationCollectionUri=https://*******.visualstudio.com/
2019-01-31T11:59:00.2354839Z ##[debug]NuGet.OverwritePackagingCollectionUrl=undefined
2019-01-31T11:59:00.2359721Z ##[debug]allowpackageconflicts=false
2019-01-31T11:59:00.2363254Z ##[debug]includenugetorg=true
2019-01-31T11:59:00.2364255Z ##[debug]nocache=false
2019-01-31T11:59:00.2365628Z ##[debug]disableParallelProcessing=false
2019-01-31T11:59:00.2366899Z ##[debug]nugetconfigpath=D:\Agents\agent2\_work\7\s
2019-01-31T11:59:00.2368053Z ##[debug]searchpatternpush=D:\Agents\agent2\_work\7\a\**\*.nupkg;!D:\Agents\agent2\_work\7\a\**\*.symbols.nupkg
2019-01-31T11:59:00.2369236Z ##[debug]selectorconfig=select
2019-01-31T11:59:00.2370567Z ##[debug]solution=D:\Agents\agent2\_work\7\s\***.sln
2019-01-31T11:59:00.2371782Z ##[debug]verbositypush=Detailed
2019-01-31T11:59:00.2372978Z ##[debug]verbosityrestore=Detailed
2019-01-31T11:59:00.2375026Z ##[debug]Agent.Version=2.144.2
2019-01-31T11:59:00.2376273Z ##[debug]SYSTEM_TASKINSTANCEID=ed931f23-ebb6-5c8f-76f6-22074363b910
2019-01-31T11:59:00.2376506Z ##[debug]SYSTEM_JOBID=fd490c07-0b22-5182-fac9-6d67fe1e939b
2019-01-31T11:59:00.2376748Z ##[debug]SYSTEM_PLANID=bbae8bc1-6b0b-4d69-a641-78beff3396f3
2019-01-31T11:59:00.2376944Z ##[debug]SYSTEM_COLLECTIONID=bf53d6fc-dae9-474d-bab0-ff9011a7e7f8
2019-01-31T11:59:00.2377400Z ##[debug]AGENT_ID=22
2019-01-31T11:59:00.2377600Z ##[debug]AGENT_MACHINENAME=*****
2019-01-31T11:59:00.2378126Z ##[debug]AGENT_NAME=*****
2019-01-31T11:59:00.2378340Z ##[debug]AGENT_JOBSTATUS=Succeeded
2019-01-31T11:59:00.2378536Z ##[debug]AGENT_OS=Windows_NT
2019-01-31T11:59:00.2378792Z ##[debug]AGENT_VERSION=2.144.2
2019-01-31T11:59:00.2383297Z ##[debug]BUILD_BUILDID=16208
2019-01-31T11:59:00.2385420Z ##[debug]BUILD_BUILDNUMBER=***-2.3_Release_CI_20190131.5
2019-01-31T11:59:00.2386393Z ##[debug]BUILD_BUILDURI=vstfs:///Build/Build/16208
2019-01-31T11:59:00.2387980Z ##[debug]BUILD_CONTAINERID=2294670
2019-01-31T11:59:00.2388501Z ##[debug]BUILD_DEFINITIONNAME=***-2.3_Release_CI
2019-01-31T11:59:00.2388944Z ##[debug]BUILD_DEFINITIONVERSION=19
2019-01-31T11:59:00.2389372Z ##[debug]BUILD_REASON=Manual
2019-01-31T11:59:00.2389735Z ##[debug]BUILD_REPOSITORY_CLEAN=true
2019-01-31T11:59:00.2390084Z ##[debug]BUILD_REPOSITORY_GIT_SUBMODULECHECKOUT=False
2019-01-31T11:59:00.2390503Z ##[debug]BUILD_REPOSITORY_NAME=***
2019-01-31T11:59:00.2390908Z ##[debug]BUILD_REPOSITORY_PROVIDER=TfsVersionControl
2019-01-31T11:59:00.2391303Z ##[debug]BUILD_SOURCEVERSION=10158
2019-01-31T11:59:00.3347398Z ##[debug]Processed: ##vso[telemetry.publish area=Packaging;feature=NuGetCommand]{"SYSTEM_TASKINSTANCEID":"ed931f23-ebb6-5c8f-76f6-22074363b910","SYSTEM_JOBID":"fd490c07-0b22-5182-fac9-6d67fe1e939b","SYSTEM_PLANID":"bbae8bc1-6b0b-4d69-a641-78beff3396f3","SYSTEM_COLLECTIONID":"bf53d6fc-dae9-474d-bab0-ff9011a7e7f8","AGENT_ID":"22","AGENT_MACHINENAME":"*****","AGENT_NAME":"*****","AGENT_JOBSTATUS":"Succeeded","AGENT_OS":"Windows_NT","AGENT_VERSION":"2.144.2","BUILD_BUILDID":"16208","BUILD_BUILDNUMBER":"***-2.3_Release_CI_20190131.5","BUILD_BUILDURI":"vstfs:///Build/Build/16208","BUILD_CONTAINERID":"2294670","BUILD_DEFINITIONNAME":"***-2.3_Release_CI","BUILD_DEFINITIONVERSION":"19","BUILD_REASON":"Manual","BUILD_REPOSITORY_CLEAN":"true","BUILD_REPOSITORY_GIT_SUBMODULECHECKOUT":"False","BUILD_REPOSITORY_NAME":"***","BUILD_REPOSITORY_PROVIDER":"TfsVersionControl","BUILD_SOURCEVERSION":"10158","command":"restore","NUGET_EXE_TOOL_PATH_ENV_VAR":"D:\\Agents\\agent2\\_work\\_tool\\NuGet\\4.7.0\\x64\\nuget.exe","searchPatternPack":"D:\\Agents\\agent2\\_work\\7\\s\\**\\*.csproj","configurationToPack":"release","versioningScheme":"off","includeReferencedProjects":false,"versionEnvVar":null,"requestedMajorVersion":"1","requestedMinorVersion":"0","requestedPatchVersion":"0","packTimezone":"utc","buildProperties":null,"basePath":null,"verbosityPack":"Detailed","includeSymbols":false,"System.TeamFoundationCollectionUri":"https://*******.visualstudio.com/","externalendpoint":null,"externalendpoints":[],"allowpackageconflicts":"false","includenugetorg":"true","nocache":"false","disableparallelprocessing":"false","nugetconfigpath":"D:\\Agents\\agent2\\_work\\7\\s","nugetfeedtype":"internal","searchpatternpush":"D:\\Agents\\agent2\\_work\\7\\a\\**\\*.nupkg;!D:\\Agents\\agent2\\_work\\7\\a\\**\\*.symbols.nupkg","selectorconfig":"select","solution":"D:\\Agents\\agent2\\_work\\7\\s\\***.sln","verbositypush":"Detailed","verbosityrestore":"Detailed","nuGetPath":"D:\\Agents\\agent2\\_work\\_tool\\NuGet\\4.7.0\\x64\\nuget.exe","nugetVersion":"4.7.0.5148"}
2019-01-31T11:59:00.3348306Z ##[debug]command=restore
2019-01-31T11:59:00.3348385Z ##[debug]Getting Packaging service access points
2019-01-31T11:59:00.3348478Z ##[debug]System.TeamFoundationCollectionUri=https://*******.visualstudio.com/
2019-01-31T11:59:00.3348571Z ##[debug]System.ServerType=Hosted
2019-01-31T11:59:00.3348678Z ##[debug]Getting credentials for local feeds
2019-01-31T11:59:00.3348748Z SYSTEMVSSCONNECTION exists true
2019-01-31T11:59:00.3348841Z ##[debug]SYSTEMVSSCONNECTION exists true
2019-01-31T11:59:00.3348953Z ##[debug]Got auth token
2019-01-31T11:59:00.3349046Z ##[debug]System.ServerType=Hosted
2019-01-31T11:59:00.3349144Z ##[debug]Agent.ProxyUrl=undefined
2019-01-31T11:59:00.3349265Z ##[debug]Getting URI for area ID B3BE7473-68EA-4A81-BFC7-9530BAAA19AD from https://*******.visualstudio.com/
2019-01-31T11:59:00.4506833Z ##[debug]Unable to get packaging URIs
2019-01-31T11:59:00.4507540Z ##[debug]{}
2019-01-31T11:59:00.4700669Z (node:9096) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Error: Error: self signed certificate in certificate chain
2019-01-31T11:59:00.4755673Z ##[section]Finishing: NuGet restore

```

### Error logs

`2019-01-30T12:21:40.4797236Z (node:10868) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Error: Error: self signed certificate in certificate chain`

**Comment by apluche** [comment]
Thank you for reaching out to us! 

Could you please post the logs for the restore task with system.debug enabled?

**Comment by KamilKh88** [comment]
```
Hi @apluche,
Following logs with system.debug flag enabled:

2019-01-31T11:58:58.0188979Z ##[debug]Evaluating condition for step: 'NuGet restore'
2019-01-31T11:58:58.0189640Z ##[debug]Evaluating: succeeded()
2019-01-31T11:58:58.0189761Z ##[debug]Evaluating succeeded:
2019-01-31T11:58:58.0189970Z ##[debug]=> True
2019-01-31T11:58:58.0190221Z ##[debug]Result: True
2019-01-31T11:58:58.0190422Z ##[section]Starting: NuGet restore
2019-01-31T11:58:58.0257487Z ==============================================================================
2019-01-31T11:58:58.0257650Z Task         : NuGet
2019-01-31T11:58:58.0257710Z Description  : Restore, pack, or push NuGet packages, or run a NuGet command. Supports NuGet.org and authenticated feeds like Package Management and MyGet. Uses NuGet.exe and works with .NET Framework apps. For .NET Core and .NET Standard apps, use the .NET Core task.
2019-01-31T11:58:58.0257771Z Version      : 2.146.0
2019-01-31T11:58:58.0257840Z Author       : Microsoft Corporation
2019-01-31T11:58:58.0257896Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkID=613747)
2019-01-31T11:58:58.0257994Z ==============================================================================
2019-01-31T11:58:58.0426005Z ##[debug]tf vc resolvePath $\*****.sln /loginType:OAuth /login:.,*** /noprompt
2019-01-31T11:58:58.4732632Z ##[debug]D:\Agents\agent2\_work\7\s\***.sln
2019-01-31T11:58:58.9557489Z ##[debug]agent.TempDirectory=D:\Agents\agent2\_work\_temp
2019-01-31T11:58:58.9603818Z ##[debug]loading inputs and endpoints
2019-01-31T11:58:58.9608887Z ##[debug]loading ENDPOINT_AUTH_PARAMETER_SYSTEMVSSCONNECTION_ACCESSTOKEN
2019-01-31T11:58:58.9618494Z ##[debug]loading ENDPOINT_AUTH_SCHEME_SYSTEMVSSCONNECTION
2019-01-31T11:58:58.9621622Z ##[debug]loading ENDPOINT_AUTH_SYSTEMVSSCONNECTION
2019-01-31T11:58:58.9623521Z ##[debug]loading INPUT_ALLOWPACKAGECONFLICTS
2019-01-31T11:58:58.9625118Z ##[debug]loading INPUT_COMMAND
2019-01-31T11:58:58.9626509Z ##[debug]loading INPUT_CONFIGURATIONTOPACK
2019-01-31T11:58:58.9628046Z ##[debug]loading INPUT_DISABLEPARALLELPROCESSING
2019-01-31T11:58:58.9629754Z ##[debug]loading INPUT_INCLUDENUGETORG
2019-01-31T11:58:58.9631201Z ##[debug]loading INPUT_INCLUDEREFERENCEDPROJECTS
2019-01-31T11:58:58.9632481Z ##[debug]loading INPUT_INCLUDESYMBOLS
2019-01-31T11:58:58.9633822Z ##[debug]loading INPUT_NOCACHE
2019-01-31T11:58:58.9635665Z ##[debug]loading INPUT_NUGETCONFIGPATH
2019-01-31T11:58:58.9637248Z ##[debug]loading INPUT_NUGETFEEDTYPE
2019-01-31T11:58:58.9638951Z ##[debug]loading INPUT_OUTPUTDIR
2019-01-31T11:58:58.9641861Z ##[debug]loading INPUT_PACKTIMEZONE
2019-01-31T11:58:58.9643336Z ##[debug]loading INPUT_PUBLISHPACKAGEMETADATA
2019-01-31T11:58:58.9644868Z ##[debug]loading INPUT_REQUESTEDMAJORVERSION
2019-01-31T11:58:58.9646595Z ##[debug]loading INPUT_REQUESTEDMINORVERSION
2019-01-31T11:58:58.9648023Z ##[debug]loading INPUT_REQUESTEDPATCHVERSION
2019-01-31T11:58:58.9649443Z ##[debug]loading INPUT_SEARCHPATTERNPACK
2019-01-31T11:58:58.9650895Z ##[debug]loading INPUT_SEARCHPATTERNPUSH
2019-01-31T11:58:58.9652134Z ##[debug]loading INPUT_SELECTORCONFIG
2019-01-31T11:58:58.9653339Z ##[debug]loading INPUT_SOLUTION
2019-01-31T11:58:58.9654489Z ##[debug]loading INPUT_TOOLPACKAGE
2019-01-31T11:58:58.9655657Z ##[debug]loading INPUT_VERBOSITYPACK
2019-01-31T11:58:58.9657016Z ##[debug]loading INPUT_VERBOSITYPUSH
2019-01-31T11:58:58.9658343Z ##[debug]loading INPUT_VERBOSITYRESTORE
2019-01-31T11:58:58.9659721Z ##[debug]loading INPUT_VERSIONINGSCHEME
2019-01-31T11:58:58.9663472Z ##[debug]loaded 28
2019-01-31T11:58:58.9676804Z ##[debug]Agent.ProxyUrl=undefined
2019-01-31T11:58:58.9677762Z ##[debug]Agent.CAInfo=undefined
2019-01-31T11:58:58.9678163Z ##[debug]Agent.ClientCert=undefined
2019-01-31T11:58:58.9678530Z ##[debug]Agent.SkipCertValidation=undefined
2019-01-31T11:58:59.0501091Z ##[debug]Agent.ProxyUrl=undefined
2019-01-31T11:58:59.0502753Z ##[debug]check path : D:\Agents\agent2\_work\_tasks\NuGetCommand_333b11bd-d341-40d9-afcf-b32d5ce6f23b\2.146.0\node_modules\vsts-task-tool-lib\lib.json
2019-01-31T11:58:59.0503986Z ##[debug]adding resource file: D:\Agents\agent2\_work\_tasks\NuGetCommand_333b11bd-d341-40d9-afcf-b32d5ce6f23b\2.146.0\node_modules\vsts-task-tool-lib\lib.json
2019-01-31T11:58:59.0504214Z ##[debug]system.culture=en-US
2019-01-31T11:59:00.0458309Z ##[debug]check path : D:\Agents\agent2\_work\_tasks\NuGetCommand_333b11bd-d341-40d9-afcf-b32d5ce6f23b\2.146.0\task.json
2019-01-31T11:59:00.0459054Z ##[debug]adding resource file: D:\Agents\agent2\_work\_tasks\NuGetCommand_333b11bd-d341-40d9-afcf-b32d5ce6f23b\2.146.0\task.json
2019-01-31T11:59:00.0459333Z ##[debug]system.culture=en-US
2019-01-31T11:59:00.0554308Z ##[debug]Getting NuGet
2019-01-31T11:59:00.0555178Z ##[debug]NuGetExeToolPath=D:\Agents\agent2\_work\_tool\NuGet\4.7.0\x64\nuget.exe
2019-01-31T11:59:00.2332138Z ##[debug]nugetfeedtype=internal
2019-01-31T11:59:00.2333008Z ##[debug]externalendpoints=null
2019-01-31T11:59:00.2335596Z ##[debug]command=restore
2019-01-31T11:59:00.2335940Z ##[debug]NuGetExeToolPath=D:\Agents\agent2\_work\_tool\NuGet\4.7.0\x64\nuget.exe
2019-01-31T11:59:00.2336308Z ##[debug]NuGetExeCustomLocation=undefined
2019-01-31T11:59:00.2338286Z ##[debug]searchPatternPack=D:\Agents\agent2\_work\7\s\**\*.csproj
2019-01-31T11:59:00.2339841Z ##[debug]configurationToPack=release
2019-01-31T11:59:00.2341163Z ##[debug]versioningScheme=off
2019-01-31T11:59:00.2343002Z ##[debug]includeReferencedProjects=false
2019-01-31T11:59:00.2344258Z ##[debug]versioningScheme=off
2019-01-31T11:59:00.2345417Z ##[debug]requestedMajorVersion=1
2019-01-31T11:59:00.2346446Z ##[debug]requestedMinorVersion=0
2019-01-31T11:59:00.2347489Z ##[debug]requestedPatchVersion=0
2019-01-31T11:59:00.2348513Z ##[debug]packTimezone=utc
2019-01-31T11:59:00.2348797Z ##[debug]buildProperties=null
2019-01-31T11:59:00.2349099Z ##[debug]basePath=null
2019-01-31T11:59:00.2351650Z ##[debug]verbosityPack=Detailed
2019-01-31T11:59:00.2353433Z ##[debug]includeSymbols=false
2019-01-31T11:59:00.2353973Z ##[debug]NuGet.UseLegacyFindFiles=undefined
2019-01-31T11:59:00.2354219Z ##[debug]NuGetTasks.IsHostedTestEnvironment=undefined
2019-01-31T11:59:00.2354461Z ##[debug]System.TeamFoundationCollectionUri=https://*******.visualstudio.com/
2019-01-31T11:59:00.2354839Z ##[debug]NuGet.OverwritePackagingCollectionUrl=undefined
2019-01-31T11:59:00.2359721Z ##[debug]allowpackageconflicts=false
2019-01-31T11:59:00.2363254Z ##[debug]includenugetorg=true
2019-01-31T11:59:00.2364255Z ##[debug]nocache=false
2019-01-31T11:59:00.2365628Z ##[debug]disableParallelProcessing=false
2019-01-31T11:59:00.2366899Z ##[debug]nugetconfigpath=D:\Agents\agent2\_work\7\s
2019-01-31T11:59:00.2368053Z ##[debug]searchpatternpush=D:\Agents\agent2\_work\7\a\**\*.nupkg;!D:\Agents\agent2\_work\7\a\**\*.symbols.nupkg
2019-01-31T11:59:00.2369236Z ##[debug]selectorconfig=select
2019-01-31T11:59:00.2370567Z ##[debug]solution=D:\Agents\agent2\_work\7\s\***.sln
2019-01-31T11:59:00.2371782Z ##[debug]verbositypush=Detailed
2019-01-31T11:59:00.2372978Z ##[debug]verbosityrestore=Detailed
2019-01-31T11:59:00.2375026Z ##[debug]Agent.Version=2.144.2
2019-01-31T11:59:00.2376273Z ##[debug]SYSTEM_TASKINSTANCEID=ed931f23-ebb6-5c8f-76f6-22074363b910
2019-01-31T11:59:00.2376506Z ##[debug]SYSTEM_JOBID=fd490c07-0b22-5182-fac9-6d67fe1e939b
2019-01-31T11:59:00.2376748Z ##[debug]SYSTEM_PLANID=bbae8bc1-6b0b-4d69-a641-78beff3396f3
2019-01-31T11:59:00.2376944Z ##[debug]SYSTEM_COLLECTIONID=bf53d6fc-dae9-474d-bab0-ff9011a7e7f8
2019-01-31T11:59:00.2377400Z ##[debug]AGENT_ID=22
2019-01-31T11:59:00.2377600Z ##[debug]AGENT_MACHINENAME=*****
2019-01-31T11:59:00.2378126Z ##[debug]AGENT_NAME=*****
2019-01-31T11:59:00.2378340Z ##[debug]AGENT_JOBSTATUS=Succeeded
2019-01-31T11:59:00.2378536Z ##[debug]AGENT_OS=Windows_NT
2019-01-31T11:59:00.2378792Z ##[debug]AGENT_VERSION=2.144.2
2019-01-31T11:59:00.2383297Z ##[debug]BUILD_BUILDID=16208
2019-01-31T11:59:00.2385420Z ##[debug]BUILD_BUILDNUMBER=***-2.3_Release_CI_20190131.5
2019-01-31T11:59:00.2386393Z ##[debug]BUILD_BUILDURI=vstfs:///Build/Build/16208
2019-01-31T11:59:00.2387980Z ##[debug]BUILD_CONTAINERID=2294670
2019-01-31T11:59:00.2388501Z ##[debug]BUILD_DEFINITIONNAME=***-2.3_Release_CI
2019-01-31T11:59:00.2388944Z ##[debug]BUILD_DEFINITIONVERSION=19
2019-01-31T11:59:00.2389372Z ##[debug]BUILD_REASON=Manual
2019-01-31T11:59:00.2389735Z ##[debug]BUILD_REPOSITORY_CLEAN=true
2019-01-31T11:59:00.2390084Z ##[debug]BUILD_REPOSITORY_GIT_SUBMODULECHECKOUT=False
2019-01-31T11:59:00.2390503Z ##[debug]BUILD_REPOSITORY_NAME=***
2019-01-31T11:59:00.2390908Z ##[debug]BUILD_REPOSITORY_PROVIDER=TfsVersionControl
2019-01-31T11:59:00.2391303Z ##[debug]BUILD_SOURCEVERSION=10158
2019-01-31T11:59:00.3347398Z ##[debug]Processed: ##vso[telemetry.publish area=Packaging;feature=NuGetCommand]{"SYSTEM_TASKINSTANCEID":"ed931f23-ebb6-5c8f-76f6-22074363b910","SYSTEM_JOBID":"fd490c07-0b22-5182-fac9-6d67fe1e939b","SYSTEM_PLANID":"bbae8bc1-6b0b-4d69-a641-78beff3396f3","SYSTEM_COLLECTIONID":"bf53d6fc-dae9-474d-bab0-ff9011a7e7f8","AGENT_ID":"22","AGENT_MACHINENAME":"*****","AGENT_NAME":"*****","AGENT_JOBSTATUS":"Succeeded","AGENT_OS":"Windows_NT","AGENT_VERSION":"2.144.2","BUILD_BUILDID":"16208","BUILD_BUILDNUMBER":"***-2.3_Release_CI_20190131.5","BUILD_BUILDURI":"vstfs:///Build/Build/16208","BUILD_CONTAINERID":"2294670","BUILD_DEFINITIONNAME":"***-2.3_Release_CI","BUILD_DEFINITIONVERSION":"19","BUILD_REASON":"Manual","BUILD_REPOSITORY_CLEAN":"true","BUILD_REPOSITORY_GIT_SUBMODULECHECKOUT":"False","BUILD_REPOSITORY_NAME":"***","BUILD_REPOSITORY_PROVIDER":"TfsVersionControl","BUILD_SOURCEVERSION":"10158","command":"restore","NUGET_EXE_TOOL_PATH_ENV_VAR":"D:\\Agents\\agent2\\_work\\_tool\\NuGet\\4.7.0\\x64\\nuget.exe","searchPatternPack":"D:\\Agents\\agent2\\_work\\7\\s\\**\\*.csproj","configurationToPack":"release","versioningScheme":"off","includeReferencedProjects":false,"versionEnvVar":null,"requestedMajorVersion":"1","requestedMinorVersion":"0","requestedPatchVersion":"0","packTimezone":"utc","buildProperties":null,"basePath":null,"verbosityPack":"Detailed","includeSymbols":false,"System.TeamFoundationCollectionUri":"https://*******.visualstudio.com/","externalendpoint":null,"externalendpoints":[],"allowpackageconflicts":"false","includenugetorg":"true","nocache":"false","disableparallelprocessing":"false","nugetconfigpath":"D:\\Agents\\agent2\\_work\\7\\s","nugetfeedtype":"internal","searchpatternpush":"D:\\Agents\\agent2\\_work\\7\\a\\**\\*.nupkg;!D:\\Agents\\agent2\\_work\\7\\a\\**\\*.symbols.nupkg","selectorconfig":"select","solution":"D:\\Agents\\agent2\\_work\\7\\s\\***.sln","verbositypush":"Detailed","verbosityrestore":"Detailed","nuGetPath":"D:\\Agents\\agent2\\_work\\_tool\\NuGet\\4.7.0\\x64\\nuget.exe","nugetVersion":"4.7.0.5148"}
2019-01-31T11:59:00.3348306Z ##[debug]command=restore
2019-01-31T11:59:00.3348385Z ##[debug]Getting Packaging service access points
2019-01-31T11:59:00.3348478Z ##[debug]System.TeamFoundationCollectionUri=https://*******.visualstudio.com/
2019-01-31T11:59:00.3348571Z ##[debug]System.ServerType=Hosted
2019-01-31T11:59:00.3348678Z ##[debug]Getting credentials for local feeds
2019-01-31T11:59:00.3348748Z SYSTEMVSSCONNECTION exists true
2019-01-31T11:59:00.3348841Z ##[debug]SYSTEMVSSCONNECTION exists true
2019-01-31T11:59:00.3348953Z ##[debug]Got auth token
2019-01-31T11:59:00.3349046Z ##[debug]System.ServerType=Hosted
2019-01-31T11:59:00.3349144Z ##[debug]Agent.ProxyUrl=undefined
2019-01-31T11:59:00.3349265Z ##[debug]Getting URI for area ID B3BE7473-68EA-4A81-BFC7-9530BAAA19AD from https://*******.visualstudio.com/
2019-01-31T11:59:00.4506833Z ##[debug]Unable to get packaging URIs
2019-01-31T11:59:00.4507540Z ##[debug]{}
2019-01-31T11:59:00.4700669Z (node:9096) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Error: Error: self signed certificate in certificate chain
2019-01-31T11:59:00.4755673Z ##[section]Finishing: NuGet restore

```

**Comment by apluche** [comment]
We're investigating the issue and depending on results, we'll work to push out a new version as soon as possible.

**Comment by apluche** [comment]
A new version of NuGetCommand should be available really soon. Please let us know if this fixes your issue!

**Comment by zarenner** [comment]
This issue appears to be entirely unrelated to #9446.

@KamilKh88, I'll be looking into this issue tomorrow and attempting to reproduce it. Is there anything interesting about your network environment (e.g. proxies, firewalls, etc)? Based on the fact that it's an error about a self-signed cert I would assume so, but the more details the better.

Thanks.

**Comment by KamilKh88** [comment]
Fixed in NuGet task version 2.146.1.
Thank you.
